src/SyntaxHighlighter/X86Assembly.cpp

```diff
@@ -0,0 +1,234 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "SyntaxHighlighter/X86Assembly.h"
+
+#include <QColor>
+#include <QRegularExpression>
+#include <QString>
+#include <QTextCharFormat>
+
+namespace orbit_syntax_highlighter {
+
+namespace {
+const QColor kCommentColor{0x99, 0x99, 0x99};
+const QColor kPlatformColor{0x99, 0x99, 0x99};
+const QColor kProgramCounterColor{0x61, 0x96, 0xcc};
+const QColor kStringColor{0x7e, 0xc6, 0x99};
+const QColor kOpcodeColor{0xf8, 0xc5, 0x55};
+const QColor kNumberColor{0xf0, 0x8d, 0x49};
+const QColor kRegisterColor{0x7e, 0xc6, 0x99};
+const QColor kKeywordColor{0xcc, 0x99, 0xcd};
+}  // namespace
+
+void X86Assembly::highlightBlock(const QString& code) {
+  const auto apply = [&code, this](const QString& pattern,
+                                   QRegularExpression::PatternOptions options,
+                                   const QColor& color) {
+    QTextCharFormat format{};
+    format.setForeground(color);
+
+    QRegularExpression expression{pattern, options};
+
+    for (auto it = expression.globalMatch(code); it.hasNext();) {
+      const auto match = it.next();
+      setFormat(match.capturedStart(), match.capturedLength(), format);
+    }
+  };
+
+  using PatternOption = QRegularExpression::PatternOption;
+  apply("\\s(0x)?[\\da-f]+\\b", PatternOption::CaseInsensitiveOption, kNumberColor);
```

Comment:
```suggestion
  apply(R"(\s(0x)?[\da-f]+\b)", PatternOption::CaseInsensitiveOption, kNumberColor);
```
These could be done this way. It might improve readability, but I don't have a preference. Just wanted to point out that this option exists.

Reply:
Raw string literals don't work well with the multi-line string literal below. So I opted for being consistent and have non-raw string literals everywhere because mixing them was even more confusing.

Reply:
Makes sense!